From f14a724ed1a595c70829e4b3c3ae85949145b8ca Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Sun, 22 Nov 2015 23:01:47 +0000
Subject: [PATCH 177/195] staging: etnaviv: rename 'gpu_active_list' to be just
 plain 'node'

Rather than adding to the structure size with more list heads, rename
the existing gpu_active_list node so we can re-use it.

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_dump.c | 4 ++--
 drivers/staging/etnaviv/etnaviv_gpu.c  | 7 +++----
 drivers/staging/etnaviv/etnaviv_gpu.h  | 2 +-
 3 files changed, 6 insertions(+), 7 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_dump.c b/drivers/staging/etnaviv/etnaviv_dump.c
index aadbd32..9b0b0b0 100644
--- a/drivers/staging/etnaviv/etnaviv_dump.c
+++ b/drivers/staging/etnaviv/etnaviv_dump.c
@@ -134,7 +134,7 @@ void etnaviv_core_dump(struct etnaviv_gpu *gpu)
 		    mmu_size + gpu->buffer->size;
 
 	/* Add in the active command buffers */
-	list_for_each_entry(cmd, &gpu->active_cmd_list, gpu_active_list) {
+	list_for_each_entry(cmd, &gpu->active_cmd_list, node) {
 		file_size += cmd->size;
 		n_obj++;
 	}
@@ -177,7 +177,7 @@ void etnaviv_core_dump(struct etnaviv_gpu *gpu)
 	etnaviv_core_dump_mem(&iter, ETDUMP_BUF_RING, gpu->buffer->vaddr,
 			      gpu->buffer->size, gpu->buffer->paddr);
 
-	list_for_each_entry(cmd, &gpu->active_cmd_list, gpu_active_list)
+	list_for_each_entry(cmd, &gpu->active_cmd_list, node)
 		etnaviv_core_dump_mem(&iter, ETDUMP_BUF_CMD, cmd->vaddr,
 				      cmd->size, cmd->paddr);
 
diff --git a/drivers/staging/etnaviv/etnaviv_gpu.c b/drivers/staging/etnaviv/etnaviv_gpu.c
index fad6bff..f289759 100644
--- a/drivers/staging/etnaviv/etnaviv_gpu.c
+++ b/drivers/staging/etnaviv/etnaviv_gpu.c
@@ -1080,12 +1080,11 @@ static void retire_worker(struct work_struct *work)
 	unsigned int i;
 
 	mutex_lock(&dev->struct_mutex);
-	list_for_each_entry_safe(cmdbuf, tmp, &gpu->active_cmd_list,
-				 gpu_active_list) {
+	list_for_each_entry_safe(cmdbuf, tmp, &gpu->active_cmd_list, node) {
 		if (!fence_is_signaled(cmdbuf->fence))
 			break;
 
-		list_del(&cmdbuf->gpu_active_list);
+		list_del(&cmdbuf->node);
 		fence_put(cmdbuf->fence);
 
 		for (i = 0; i < cmdbuf->nr_bos; i++) {
@@ -1229,7 +1228,7 @@ int etnaviv_gpu_submit(struct etnaviv_gpu *gpu,
 	etnaviv_buffer_queue(gpu, event, cmdbuf);
 
 	cmdbuf->fence = fence;
-	list_add_tail(&cmdbuf->gpu_active_list, &gpu->active_cmd_list);
+	list_add_tail(&cmdbuf->node, &gpu->active_cmd_list);
 
 	for (i = 0; i < submit->nr_bos; i++) {
 		struct etnaviv_gem_object *etnaviv_obj = submit->bos[i].obj;
diff --git a/drivers/staging/etnaviv/etnaviv_gpu.h b/drivers/staging/etnaviv/etnaviv_gpu.h
index 5cb0b6e..d349e41 100644
--- a/drivers/staging/etnaviv/etnaviv_gpu.h
+++ b/drivers/staging/etnaviv/etnaviv_gpu.h
@@ -153,7 +153,7 @@ struct etnaviv_cmdbuf {
 	/* target exec state */
 	u32 exec_state;
 	/* per GPU in-flight list */
-	struct list_head gpu_active_list;
+	struct list_head node;
 	/* BOs attached to this command buffer */
 	unsigned int nr_bos;
 	struct etnaviv_gem_object *bo[0];
-- 
2.6.2

